Assets/__Scripts/Map/Obstacles/BeatmapObstacleContainer.cs

```diff
@@ -111,29 +111,6 @@ public override void UpdateGridPosition()
             transform.RotateAround(rectWorldPos, transform.forward, localRotation.z);
         }
 
-        UpdateCollisionGroupsWithDuration(duration);
-    }
-
-    private void UpdateCollisionGroupsWithDuration(float duration)
-    {
-        var chunkStart = ChunkID;
-
-        var chunkEnd = (int)((obstacleData._time + duration) / Intersections.ChunkSize);
-
-        if (chunkStart > chunkEnd)
-        {
-            // me and the boys casually flipping shit
-            (chunkStart, chunkEnd) = (chunkEnd, chunkStart);
-        }
-
-        var range = Enumerable.Range(chunkStart, Mathf.Max(chunkEnd - chunkStart, 1));
-
-        foreach (var collider in colliders)
-        {
-            var unregistered = Intersections.UnregisterColliderFromGroups(collider);
-            collider.CollisionGroups.Clear();
-            collider.CollisionGroups.AddRange(range);
-            if (unregistered) Intersections.RegisterColliderToGroups(collider);
-        }
```

Comment:
Wouldn't this removal affect selecting objects that are behind long obstacles? What about negative duration walls?

Reply:
Because of the change where we don't stop after checking a single chunk but check all the loaded chunks anything on screen should get checked so adding the obstacle to multiple chunks isn't necessary.

Reply:
And those cases were broken before this change which is exactly why I've changed it. This system didn't work.